platformio.ini

```diff
@@ -38,6 +38,13 @@ extra_configs =
 # The 'common' values are used for most Marlin builds
 #
 [common]
+build_flags        = -fmax-errors=5 -g3 -D__MARLIN_FIRMWARE__ -fmerge-constants -fno-signed-zeros -ffinite-math-only
+extra_scripts      =
```

Comment:
Should it be in common or AVR only ? I expect CPU with FPU not to require this and larger ARM CPU could support the overhead.

Reply:
There's only a benefit if there's a need to use it. At this time we don't need a NAN since any value will do in its place. And, we don't need "infinity" for any reason either. At the bottom, Marlin is founded on an algorithm developed for plotters in 1962, so we can factor out a lot of higher level maths that we don't use. Maybe we can also dump trigonometry for certain configurations. Hmm….